src/components/Tabs/TabItem.tsx

```diff
@@ -0,0 +1,49 @@
+import React, { FC, ReactElement, useState } from "react";
+import { Badge, BadgeProps } from "@components/Badge";
+import { motion } from "framer-motion";
+import { merge } from "@utilities/merge";
+
+export type TabITemProps = {
+    decorator?: ReactElement;
+    badge?: BadgeProps;
+    label: string;
+    onClick: () => void;
+    active: boolean;
+    disabled: boolean;
+};
+
+export const TabItem: FC<TabITemProps> = (props: TabITemProps) => {
+    const [isVisible, setIsVisible] = useState(false);
```

Comment:
The component shouldn't be controlled internally
```
const [selectId, setSelectedId] = useState("tab-1")

<Tabs selectedId={selectId} onChange={setSelectedId}>
	<TabItem label="Tab 1" id="tab-1">Content of tab 1</TabItem>
	<TabItem label="Tab 2" id="tab-2">Content of tab 2</TabItem>
<Tabs>
```

So most of the logic live in the Tabs and outside